### PR TITLE
adding metadata tests back to integration tests

### PR DESCRIPTION
**Pull request by sharifelgamal**

### CODE REVIEW

File: integration/integration_test.go [hunk 1]
```diff
@@ -93,16 +102,24 @@ func TestRun(t *testing.T) {
 			"file=context/foo",
 			"file3=context/b*",
 		},
+		"Dockerfile_test_multistage": {"file=/foo2"},
 	}
 
 	bucketContextTests := []string{"Dockerfile_test_copy_bucket"}
 
+	testsToIgnore := []string{"Dockerfile_test_user_run"}
```
Comment (priyawadhwa) [inline integration/integration_test.go]: Why are we ignoring this test?

File: pkg/executor/executor.go [hunk 2]
```diff
@@ -99,6 +99,9 @@ func DoBuild(dockerfilePath, srcContext, snapshotMode string, args []string) (na
 			return nil, nil, err
 		}
 		imageConfig, err := sourceImage.ConfigFile()
+		if baseImage == constants.NoBaseImage {
+			imageConfig.Config.Env = []string{"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"}
```
Comment (priyawadhwa) [inline pkg/executor/executor.go]: Could we add this to the constants package?

Comment (priyawadhwa) [inline pkg/executor/executor.go]: Could we add this to the constants package?

File: integration/dockerfiles/Dockerfile_test_expose [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FROM gcr.io/google-appengine/debian9
+EXPOSE 80
+EXPOSE 81/udp
+ENV protocol tcp
+EXPOSE 82/$protocol
+ENV port 83
+EXPOSE $port/udp
+EXPOSE $port/$protocol
```
Comment (priyawadhwa) [inline integration/dockerfiles/Dockerfile_test_expose]: nit: newline

File: integration/dockerfiles/Dockerfile_test_user_run [hunk 4]
```diff
@@ -17,3 +17,5 @@ RUN useradd testuser
 RUN groupadd testgroup
 USER testuser:testgroup
 RUN echo "hey" > /tmp/foo
+USER testuser:1001
+RUN echo "hey2" >> /tmp/foo
```
Comment (priyawadhwa) [inline integration/dockerfiles/Dockerfile_test_user_run]: nit: newline

File: integration/integration_test.go (hunk 1 above)
Comment (tejal29) [reply integration/integration_test.go]: just add the bug as a todo comment.

File: pkg/commands/user.go [hunk 5]
```diff
@@ -47,43 +46,15 @@ func (r *UserCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.Bu
 		}
 	}
 
-	// Lookup by username
-	userObj, err := user.Lookup(userStr)
+	_, _, err = util.GetUserFromUsername(userStr, groupStr)
```
Comment (tejal29) [inline pkg/commands/user.go]: Little bit confused here.. 
Not sure if its pertaining to this PR. 
But we are ignoring the resolved user and group and returning the input userStr:grpStr correct?
Is this command to just see if user and group can be resolved?

File: pkg/commands/user.go [hunk 6]
```diff
@@ -47,43 +46,15 @@ func (r *UserCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile.Bu
...
-
-	uid := userObj.Uid
-	if group != nil {
-		uid = uid + ":" + group.Gid
+		userStr = userStr + ":" + groupStr
 	}
-
-	logrus.Infof("Setting user to %s", uid)
-	config.User = uid
+	config.User = userStr
```
Comment (tejal29) [inline pkg/commands/user.go]: Initially this was set to uid:gid.
Now its being set to userStr:groupStr.
Is this intended ?

File: pkg/executor/executor.go (hunk 2 above)
Comment (sharifelgamal) [reply pkg/executor/executor.go]: Done

File: pkg/commands/user.go (hunk 6 above)
Comment (sharifelgamal) [reply pkg/commands/user.go]: Yes, that's how docker does it and we want to mimic what docker does as closely as possible.

**Pull request merged**